spec/validators/chat_allow_uploads_validator_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe ChatAllowUploadsValidator do
+  it "always returns true if setting the value to false" do
+    validator = described_class.new
+    expect(validator.valid_value?("f")).to eq(true)
+  end
+
+  context "when secure media is enabled" do
+    before do
+      SiteSetting.chat_allow_uploads = false
+      enable_secure_media
+    end
+
+    after { ENV["DISCOURSE_ALLOW_UNSECURE_CHAT_UPLOADS"] = nil }
+
+    it "does not allow chat uploads to be enabled" do
+      validator = described_class.new
+      expect(validator.valid_value?("t")).to eq(false)
+      expect(validator.error_message).to eq(
+        I18n.t("site_settings.errors.chat_upload_not_allowed_secure_media"),
+      )
+    end
+
+    it "does allow chat uploads to be enabled if DISCOURSE_ALLOW_UNSECURE_CHAT_UPLOADS is set" do
+      ENV["DISCOURSE_ALLOW_UNSECURE_CHAT_UPLOADS"] = "t"
```

Comment:
Should we specify the value that the ENV should set? Another spec sets it to `true` while this spec sets it to `t`.

Reply:
Irrelevant now I have changed to GlobalSetting